src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -249,7 +249,12 @@ private DocWriteRequest<?> maybeAddExternalVersioning(
   ) {
     if (!config.isDataStream() && !config.shouldIgnoreKey(record.topic())) {
       request.versionType(VersionType.EXTERNAL);
-      request.version(record.kafkaOffset());
+      if (config.hasExternalVersionHeader()) {
+        request.version((Long) record.headers().lastWithName(
+                config.externalVersionHeader()).value());
```

Comment:
@rjosal-indeed 
Thanks for putting changes in PR.

While our testing, we faced this issue, where kafka header values were being treated as bytes instead of string, due to which we were getting this exception:

`java.lang.ClassCastException: class java.lang.Byte cannot be cast to class java.lang.Long (java.lang.Byte and java.lang.Long are in module java.base of loader 'bootstrap')`

Some people may resolve it by using an SMT but we can also generalise the changes. 

Few suggestions:

- Keep `request.version` method inside a try/catch block and throwing error wrapped under `ConnectException`. This can ensure graceful failure of the connector.

- Generalise `Object` to `Long` conversion (like this):
`request.version(Long.parseLong(
            String.valueOf(
                record.headers().lastWithName(
                    config.externalVersionHeader()).value().toString())));`

- Change the base branch of the PR to `10.0.x`.
                         
Let us know if you have any comments. Thanks.

Reply:
Hey thanks for testing and reviewing!  I actually had a comment above about using `SimpleHeaderConverter#fromConnectHeader` to help ensure parsing without assuming schema - can I request guidance/confirmation to use the parseLong approach, or the formatted SimpleHeaderCoverter utility?

Reply:
@sp-gupta  I worked for awhile on trying to backport this to 10.0.x and ended up failing.  10.0.x is missing a critical element in external versioning usage for offset which was extended here.  I would have to backport significant new functionality from 11.0.x to get it to work, which presumably wasn't backported for some reason, so feels a bit risky.  I can successfully apply this patch to 11.0.x.

I wonder generally though if backporting is expected for new features?  CONTRIBUTING.md indicates this is generally for bugfixes
> If your code changes are essentially bug fixes that make sense to backport to existing releases

This feature I'd expect to have documentation updates on old releases as well as making initial release announcements retroactively incomplete.  Thoughts?

Reply:
Hey @rjosal-indeed 
Thanks for the changes. 

As per discussion with the team, we can change the base branch to 11.0.x and we don't need to backport the changes to 10.0.x (since this is not a bug fix).

Yes, Backporting is **not** expected for new features. Documentation updates should cover informing customers about this feature.

Please go ahead and change the base branch to 11.0.x, after which I'll approve the PR.